tools/chplcheck/src/rules.py

```diff
@@ -323,3 +323,43 @@ def variables(node):
 
         for unused in indices.keys() - uses:
             yield indices[unused]
+
+    @driver.advanced_rule
+    def SimpleDomainAsRange(context: Context, root: AstNode):
+        """
+        Warn for simple domains in loops that can be ranges.
+        """
+
+        def is_range_like(node: AstNode):
+            """
+            a node is range like if its a range, a `count` expr with a
+            range-like on the lhs, a `by` expr with a range-like on the lhs, or
+            an `align` expr with a range-like on the lhs
+            """
+
+            if isinstance(node, Range):
+                return True
+            if (
+                isinstance(node, OpCall)
+                and node.is_binary_op()
+                and (
+                    node.op() == "#"
+                    or node.op() == "by"
+                    or node.op() == "align"
+                )
+            ):
+                return is_range_like(list(node.actuals())[0])
```

Comment:
Can we just add an `actual(idx)` getter... This is really unfortunate.
Alternatively, `next(node.actuals())` should work.

Reply:
I think its perfectly reasonable to add a `actual(idx)` getter. I'll add to this PR